File: swift/DesignPattern.playground/Pages/AbstractFactoryPattern.xcplaygroundpage/Sources/USBeverageFactory.swift
import Foundation

open class USBeverageFactory: BeverageFactory {
    
    public init() {}
    
    public func createBlackTea() -> BlackTea? {
        return CeylonBlackTea()
    }
    
    public func createGreenTea() -> GreenTea? {
        return GyokuroGreenTea()
    }
}

File: kotlin/DesignPattern/src/main/kotlin/abstractfactorypattern/source/USBeverageFactory.kt
package abstractfactorypattern.source

class USBeverageFactory: BeverageFactory {

    override fun createBlackTea(): BlackTea {
        return CeylonBlackTea()
    }

    override fun createGreenTea(): GreenTea {
        return GyokuroGreenTea()
    }
}